resources/assets/js/store/modules/permissions.js

```diff
@@ -0,0 +1,162 @@
+import api from 'plugins/http/api';
+
+/**
+ * Simple interface for interacting with user roles and permissions
+ * 
+ * permissions is an array with the following structure
+ * 
+[
+		{
+			"name": "Create Subsites",
+			"slug": "subsite.create",
+			"roles": [
+				"site.owner"
+			]
+		},
+		...
+ */
+const state = {
+	roles: [],
+	currentRole: '',
+	globalRole: '',
+};
+
+const getters = {
+	getCurrentRole(state, getters) {
+		return state.currentSiteRole;
+	},
+
+	getRoles(state, getters) {
+		return state.roles;
+	},
+
+	/**
+	 * can the user perform the requested action
+	 * if the user is a global admin then let them do anything
+	 * 
+	 * @param {object} state - the vuex state
+	 * @param {string} the permission to check i.e. subsite.edit, subsite.create
+	 * @returns {bool} true or false
+	 */
+	canUser: (state, getters) => (permissionSlug) => {
+
+		let permitted = false;
+		
+		// if the user has the global role of admin then then they can do anything
+		if (state.globalRole === 'admin') {
+			return true;
+		}
+		const matchedRole = state.roles.find(function(value) {
```

Comment:
would make more sense to me as state.permissions.find(function(value) { .... 
as you are looking for a permission in a list of permissions, then looking up roles.

Reply:
Agree, changed state.roles to state.permissions